src/server/pkg/collection/iterator.go

```diff
@@ -22,84 +23,71 @@ type Options struct {
 // DefaultOptions are the default sort options when iterating through etcd key/values.
 var DefaultOptions = &Options{etcd.SortByCreateRevision, etcd.SortDescend, false}
 
-type etcdIterator interface {
-	next() ([]*mvccpb.KeyValue, error)
-	done() bool
-}
-
-type iterator struct {
-	c           *readonlyCollection
-	prefix      string
-	limitPtr    *int64
-	opts        *Options
-	compareFunc func(*mvccpb.KeyValue, *mvccpb.KeyValue) int
-	last        bool
-}
-
-func newEtcdIterator(c *readonlyCollection, prefix string, limitPtr *int64, opts *Options) etcdIterator {
+func iterate(c *readonlyCollection, prefix string, limitPtr *int64, opts *Options, f func(*mvccpb.KeyValue) error) error {
```

Comment:
This is a bit pedantic but how would you feel about renaming the functions in this file from `iterator` to `list` to match a little more with the other ones? I also think this function in particular is close enough to `listF` that you could just rename it to `listF` it seems like the only difference is that the `readonlyCollection` is an argument rather than the receiver.

Also if you want you could drop the `F`s from the ends of these functions, I only named them that because I was gradually migrating from code that had a bunch of functions called `List` already.